web-api/src/mind.rs

```diff
@@ -231,6 +287,121 @@ where
         .map_err(Into::into)
 }
 
+// function that assigns a score to a vector of documents based on the user's interests
+// score is equal to 1 if the document is interesting to the user, 0 otherwise
+fn score_documents(documents: &[&Document], user_interests: &UserInterests) -> Vec<f32> {
+    documents
+        .iter()
+        .map(|document| {
+            if document.is_interesting(user_interests) {
+                1.0
+            } else {
+                0.0
+            }
+        })
+        .collect()
+}
+
+fn write_array<T, S, D>(writer: impl io::Write, array: &ndarray::ArrayBase<S, D>) -> io::Result<()>
+where
+    T: Clone + npyz::AutoSerialize,
+    S: ndarray::Data<Elem = T>,
+    D: ndarray::Dimension,
+{
+    let shape = array.shape().iter().map(|&x| x as u64).collect::<Vec<_>>();
+    let c_order_items = array.iter();
+
+    let mut writer = npyz::WriteOptions::new()
+        .default_dtype()
+        .shape(&shape)
+        .writer(writer)
+        .begin_nd()?;
+    writer.extend(c_order_items)?;
+    writer.finish()
+}
+
+/// Runs the persona-based MIND benchmark. We load the user's preferences from the json file and
+/// then use the user's preferences to select the documents that are relevant to the user.
+/// The documents are then used to prepare reranker. The reranker is then used to rerank the
+/// randomly selected documents. Then NDCG is calculated.
+fn run_persona_based_benchmark() -> Result<(), Error> {
+    // define random thread
+    let mut rng = thread_rng();
+    // define how many documents to sample
+    let amount_of_doc_used_to_prepare = 1;
+
+    // define how many documents to rerank
+    let amount_of_doc_used_to_rerank = 30;
+
+    //probability of a document being clicked if it is interesting to the user
+    let click_probability = 0.5;
+
+    //define how many iterations to perform
+    let iterations = 10;
+
+    // define how many documents to use to calculate NDCG
+    let nranks = vec![3, 5];
```

Comment:
:art: looks like all those values could go into a config struct, maybe not now, but in a later pr to make it easer to setup different test cases.

Reply:
Good idea, I will create a such struct